### PR TITLE
chore(ci): add pipeline scans

### PR DESCRIPTION
**Pull request by sc979**
## Description

Add main scans per project and versions
Add pipeline scans per project and version 

**Fixes** # (https://centreon.atlassian.net/browse/MON-19174)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 21.10.x
- [x] 22.04.x
- [x] 22.10.x
- [x] 23.04.x
- [x] 23.10.x (master)

### CODE REVIEW

File: .github/actions/veracode-get-cache/action.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        sudo ./aws/install
+      shell: bash
+
+    - name: Get baseline file
+      run: |
+        TARGET_S3="s3://centreon-veracode-reports/${{ inputs.module_name }}/${{ inputs.target_branch_name }}/${{ inputs.baseline_filename }}"
+        aws s3 ls "$TARGET_S3" || RESULT=$( echo $? )
+        if [[ $RESULT -ne 0 ]]; then
+          echo "[DEBUG] - File not found in the bucket"
+          touch /tmp/${{ inputs.baseline_filename }}
```
Comment (sc979) [inline .github/actions/veracode-get-cache/action.yml:30]:
```suggestion
          touch /tmp/${{ inputs.baseline_filename }}
          chmod 660  /tmp/${{ inputs.baseline_filename }}
```

File: .github/actions/veracode-generate-archive/action.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+name: "generate-veracode-binary"
```
Comment (kduret) [inline .github/actions/veracode-generate-archive/action.yml:1]: veracode-generate-archive

File: .github/actions/veracode-generate-archive/action.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+description: "Prepare binary to be analyzed"
```
Comment (kduret) [inline .github/actions/veracode-generate-archive/action.yml:2]:
```suggestion
description: "Prepare archive to be analyzed"
```

File: .github/actions/veracode-generate-archive/action.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      run: |
+        cd ${{ inputs.module_directory }}
+        zip -rq "../${{ inputs.module_name }}-${{ inputs.target_branch_name }}.zip" *
+      shell: bash
+
+    - name: DEBUG - Check location
+      run: ls -la #TODO: to remove
+      shell: bash
+
+    - uses: actions/upload-artifact@v3
```
Comment (kduret) [inline .github/actions/veracode-generate-archive/action.yml:47]: prefer usage of actions/cache except if you need it outside the workflows

File: .github/workflows/veracode-pipeline-scan.yml [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+          baseline_file: ${{ env.BASELINE_FILE }}
+
+      - name: Move baseline file
+        run: mv results.json /tmp
+
+      - name: Get current time
+        run: echo "$(date +'%Y-%m-%d %H:%M:%S')"
+
+      - name: Cache baseline
+        if: env.HEAD_REF == 'develop' || env.HEAD_REF == 'dev-[2-9][0-9].[0-9][0-9].x' || env.HEAD_REF == 'master' || env.HEAD_REF == '[2-9][0-9].[0-9][0-9].x' || env.HEAD_REF == 'AT-94-pipeline-scan'
```
Comment (kduret) [inline .github/workflows/veracode-pipeline-scan.yml:95]:
```suggestion
        if: github.event_name == 'pull_request'
```

File: .github/actions/veracode-set-cache/action.yml
Comment (kduret) [inline .github/actions/veracode-set-cache/action.yml:1]: if this action will not be used by multiple workflows, integrate it directly in the steps of `.github/workflows/veracode-pipeline-scan.yml`

File: .github/workflows/veracode-release-branch-scan.yml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  MODULE_NAME: centreon-gorgone
+  CRITICALITY: "VeryLow"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Generate binary
```
Comment (kduret) [inline .github/workflows/veracode-release-branch-scan.yml:21]:
```suggestion
      - name: Generate archive
```

**Pull request closed without merging**